packages/belos/src/BelosSolverFactory_Belos.hpp

```diff
@@ -62,6 +62,14 @@ class BelosSolverFactory : public Impl::SolverFactoryParent<double,MultiVec<doub
     };
 };
 
+class BelosFloatSolverFactory : public Impl::SolverFactoryParent<float,MultiVec<float>,Operator<float>>
```

Comment:
Does this need to be guarded? The complex stuff has a `#ifdef HAVE_TEUCHOS_COMPLEX`.

Reply:
Possibly, but I am not sure what to guard it with.  When I wrapped the Factory.cpp test with `#ifdef Trilinos_ENABLE_FLOAT `, that test did not build, even though I had enabled that option.

Reply:
@hkthorn ??

Reply:
I would say that "float" does not need to be guarded.  There are basic tests of LAPACK/BLAS/SDM in Teuchos that are based on float and are not guarded.  If the linear algebra stack supports float, then it should be fine.  Tpetra is controlling which scalar types it is enabling, like float, but this is not the only linear algebra that Belos can work with.